### PR TITLE
Refactor setup.rb

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

**Comment by ehelms** [comment]
This comes with 1 interface change (how installation type is specified) and includes a lot more changes to setup.rb for all the "cases" it was handling. For some changes, the refactoring was more obvious and clean, for others I was a little unsure of how to go but picked a design and went with it.

File: Vagrantfile [hunk 1]
```diff
@@ -31,12 +31,12 @@ module KatelloDeploy
     { :name => 'centos6-2.1', :shell => "#{INSTALL_SHELL} --version=2.1" }.merge(BASE_BOXES.fetch(:centos6)),
     { :name => 'centos6-2.2', :shell => "#{INSTALL_SHELL} --version=2.2" }.merge(BASE_BOXES.fetch(:centos6)),
     { :name => 'centos6-bats', :shell => BATS_SHELL }.merge(BASE_BOXES.fetch(:centos6)),
-    { :name => 'centos6-devel', :shell => "#{INSTALL_SHELL} --devel" }.merge(BASE_BOXES.fetch(:centos6)),
+    { :name => 'centos6-devel', :shell => "#{INSTALL_SHELL} --install_type='devel'" }.merge(BASE_BOXES.fetch(:centos6)),
```
Comment (stbenjam) [inline Vagrantfile]: --install-type, right?

Comment (ehelms) [inline Vagrantfile]: Good eye, fixed.

**Comment by ehelms** [comment]
I found some issues testing that I had not accounted for but those should be good to go now.

**Comment by ehelms** [comment]
@stbenjam Travis is green now if you want to re-review.

File: setup.rb [hunk 2]
```diff
@@ -80,242 +66,43 @@ def system_with_debug(*args)
...
+options[:os] ||= operating_system.detect
+operating_system.supported?(options[:os])
+
+KatelloDeploy::Processors::KojiTaskProcessor.process(options[:koji_task])
+KatelloDeploy::Processors::ModulePullRequestProcessor.process(options[:module_prs], File.expand_path(File.dirname(__FILE__)))
+
+repositories = KatelloDeploy::Repositories.new(
+  :katello_version => options[:version],
+  :os_version => operating_system.version(options[:os]),
+  :distro => operating_system.distro(options[:os])
```
Comment (stbenjam) [inline setup.rb]: This should be accessed as a hash, no?

```
./setup.rb:90:in `distro': wrong number of arguments (1 for 0) (ArgumentError)
	from ./setup.rb:90
```

Comment (ehelms) [inline setup.rb]: That explains something I was seeing -- updated the distro method.

**Comment by stbenjam** [comment]
@ehelms This needs rebasing

**Comment by ehelms** [comment]
@stbenjam done

**Comment by stbenjam** [comment]
Looks good to me, just rubocop :cop: it looks like..

APT

**Pull request merged**